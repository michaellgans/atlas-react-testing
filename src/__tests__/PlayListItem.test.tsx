// Test for PlayListItem Component

// Imports
import { render } from "@testing-library/react";
import { PlayListItem } from "@/components/PlayListItem";
import { AppContext } from "@/components/ContextProvider";
import { expect, it, vi } from "vitest";

// Creates Mock Context Provider
// MOCKS ALL DATA RETURNED BY CONTEXT PROVIDER!
const mockContextValue = {
  songs: [
    { title: "Test Song 1", artist: "Test Artist 1", duration: "3:00", cover: "cover 1" },
    { title: "Test Song 2", artist: "Test Artist 2", duration: "4:00", cover: "cover 2" }
  ],
  // First song is selected in the test
  currentSong: 0,
  // Mock setCurrentSong useState function
  setCurrentSong: vi.fn(),
};

it("should render the PlayListItem component and match the snapshot", () => {
  const { asFragment } = render(
    <AppContext.Provider value={mockContextValue}>
      <PlayListItem songTitle="Test Song 1" artist="Test Artist 1" playTime="3:00" />
    </AppContext.Provider>
  );

  // Take Snapshot
  expect(asFragment()).toMatchSnapshot();
});
