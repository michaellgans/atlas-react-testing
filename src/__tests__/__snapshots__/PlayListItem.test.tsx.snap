// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PlayListItem component and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="{currentplay-list-item mb-1 flex h-10 cursor-pointer items-center justify-between rounded-md text-sm font-medium bg-hover dark:bg-dark-hover active:bg-active dark:hover:bg-dark-hover dark:active:bg-dark-active"
  >
    <div
      class="song-info"
    >
      <div
        class="song-title pb-0.5"
      >
        Song A
      </div>
      <div
        class="genre text-muted-text dark:text-dark-muted-text"
      >
        Artist A
      </div>
    </div>
    <div
      class="play-time text-muted-text dark:text-dark-muted-text"
    >
      3:45
    </div>
  </div>
</DocumentFragment>
`;
